src/librustdoc/clean/inline.rs

```diff
@@ -435,7 +432,7 @@ crate fn build_impl(
 
     debug!("build_impl: impl {:?} for {:?}", trait_.def_id(), for_.def_id());
 
-    ret.push(clean::Item::from_def_id_and_parts(
+    let mut item = clean::Item::from_def_id_and_parts(
```

Comment:
I like to distinguish when things are being overriden by rustdoc (as opposed to what most tools will see).
```suggestion
    let mut what_rustc_thinks = clean::Item::from_def_id_and_parts(
```
I guess it doesn't make as much sense here because you used mutability instead of record update syntax ... up to you whether you want to change it.